src/Ellipsoid3d.elm

```diff
@@ -0,0 +1,444 @@
+    Frame3d.yAxis (axes ellipsoid)
+
+
+{-| Get the Z axis of an ellipsoid.
+-}
+zAxis : Ellipsoid3d units coordinates -> Axis3d units coordinates
+zAxis ellipsoid =
+    Frame3d.zAxis (axes ellipsoid)
+
+
+{-| Get the radius of an ellipsoid along its X axis.
+
+    Ellipsoid3d.xRadius exampleEllipsoid
+    --> Length.meters 5
+
+-}
+xRadius : Ellipsoid3d units coordinates -> Quantity Float units
+xRadius (Types.Ellipsoid3d ellipsoid) =
+    ellipsoid.xRadius
+
+
+{-| Get the radius of an ellipsoid along its Y axis.
+
+    Ellipsoid3d.yRadius exampleEllipsoid
+    --> Length.meters 3
+
+-}
+yRadius : Ellipsoid3d units coordinates -> Quantity Float units
+yRadius (Types.Ellipsoid3d ellipsoid) =
+    ellipsoid.yRadius
+
+
+{-| Get the radius of an ellipsoid along its Z axis.
+
+    Ellipsoid3d.zRadius exampleEllipsoid
+    --> Length.meters 4
+
+-}
+zRadius : Ellipsoid3d units coordinates -> Quantity Float units
+zRadius (Types.Ellipsoid3d ellipsoid) =
+    ellipsoid.zRadius
+
+
+{-| Get the direction of the ellipsoid's X axis.
+-}
+xDirection : Ellipsoid3d units coordinates -> Direction3d coordinates
+xDirection ellipsoid =
+    Frame3d.xDirection (axes ellipsoid)
+
+
+{-| Get the direction of the ellipsoid's Y axis.
+-}
+yDirection : Ellipsoid3d units coordinates -> Direction3d coordinates
+yDirection ellipsoid =
+    Frame3d.yDirection (axes ellipsoid)
+
+
+{-| Get the direction of the ellipsoid's Z axis.
+-}
+zDirection : Ellipsoid3d units coordinates -> Direction3d coordinates
+zDirection ellipsoid =
+    Frame3d.zDirection (axes ellipsoid)
+
+
+{-| Get the volume of an ellipsoid.
+-}
+volume : Ellipsoid3d units coordinates -> Quantity Float (Cubed units)
+volume ellipsoid =
+    let
+        axesProduct =
+            xRadius ellipsoid
+                |> Quantity.times (yRadius ellipsoid)
+                |> Quantity.times (zRadius ellipsoid)
+    in
+    Quantity.multiplyBy (4 / 3 * pi) axesProduct
+
+
+{-| Get the minimal bounding box containing a given ellipsoid.
+
+    Ellipsoid3d.boundingBox exampleSphere
+    --> BoundingBox3d.from
+    -->     (Point3d.meters -3 -2 -1)
+    -->     (Point3d.meters 7 4 7)
+
+-}
+boundingBox : Ellipsoid3d units coordinates -> BoundingBox3d units coordinates
+boundingBox ellipsoid =
+    BoundingBox3d.xyz
+        (signedDistanceAlong Axis3d.x ellipsoid)
+        (signedDistanceAlong Axis3d.y ellipsoid)
+        (signedDistanceAlong Axis3d.z ellipsoid)
+
+
+{-| Check if an ellipsoid contains a given point.
+-}
+contains : Point3d units coordinates -> Ellipsoid3d units coordinates -> Bool
+contains point ellipsoid =
+    let
+        localFrame =
+            axes ellipsoid
+
+        x =
+            Point3d.xCoordinateIn localFrame point
+
+        y =
+            Point3d.yCoordinateIn localFrame point
+
+        z =
+            Point3d.zCoordinateIn localFrame point
+
+        c1 =
+            Quantity.squared x
+                |> Quantity.ratio (Quantity.squared (xRadius ellipsoid))
```

Comment:
I think this ratio looks backwards...`Quantity.ratio a b` is `a / b`, but using it in a pipeline like this means that you're doing `r_x^2 / x^2` instead of `x^2 / r_x^2`.

Reply:
wow! you're right, I mixed up the ratio! Now it's fixed :smile: https://github.com/ianmackenzie/elm-geometry/pull/142/commits/178eb76bb5ead493b9693e25e8090bb9b931a072